### PR TITLE
TELCODOCS-852 - Document New Features (4.12 Telco-RAN)

### PR DESCRIPTION
**Pull request by slovern**
[TELCODOCS-852](https://issues.redhat.com//browse/TELCODOCS-852) - Document New Features (4.12 Telco-RAN)

Version(s): 4.12

Issue: https://issues.redhat.com/browse/TELCODOCS-852

Link to docs preview: https://54503--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-12-release-notes.html#ocp-4-12-new-features-and-enhancements

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: release_notes/ocp-4-12-release-notes.adoc [hunk 1]
```diff
@@ -884,6 +884,55 @@ In {product-title} {product-version}, you can use the {factory-prestaging-tool}
...
+For more information, see xref:../scalability_and_performance/optimizing-cpu-usage.adoc[Optimizing CPU usage with mount namespace encapsulation]. 
+
+[id="ocp-4-12-ztp-sno-workload-partitioning"]
+==== Changing the workload partitioning CPU set in {sno} clusters that are deployed with GitOps ZTP
+
+You can configure the workload partitioning CPU set in {sno} clusters that you deploy with GitOps ZTP.
+To do this, you specify cluster management CPU resources with the `cpuset` field of the `SiteConfig` custom resource (CR) and the `reserved` field of the group `PolicyGenTemplate` CR.
+The value that you set for `cpuset` should match the value set in the cluster `PerformanceProfile` CR `.spec.cpu.reserved` field for workload partitioning.
+
+For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-reference-cluster-configuration-for-vdu.adoc#ztp-workload-partitioning-sno_sno-configure-for-vdu#ztp-using-hub-cluster-templates-with-talm_ztp-advanced-policy-config[Workload partitioning in {sno} with GitOps ZTP].
```
Comment (aireilly) [inline release_notes/ocp-4-12-release-notes.adoc:911]:
```suggestion
For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.adoc#ztp-using-hub-cluster-templates_ztp-advanced-policy-config[Using hub templates in PolicyGenTemplate CRs]. 
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 2]
```diff
@@ -884,6 +884,55 @@ In {product-title} {product-version}, you can use the {factory-prestaging-tool}
...
+
+A standard host operating system uses systemd to constantly scan all mount namespaces: both the standard Linux mounts and the numerous mounts that Kubernetes uses to operate. The current implementation of Kubelet and CRI-O both use the top-level namespace for all container and Kubelet mount points. Encapsulating these container-specific mount points in a private namespace reduces systemd overhead and enhances CPU performance. Encapsulation can also improve security, by storing Kubernetes-specific mount points in a location safe from inspection by unprivileged users. 
+
+For more information, see xref:../scalability_and_performance/optimizing-cpu-usage.adoc[Optimizing CPU usage with mount namespace encapsulation].
```
Comment (aireilly) [inline release_notes/ocp-4-12-release-notes.adoc:902]:
```suggestion
For more information, see xref:../scalability_and_performance/optimizing-cpu-usage.adoc#optimizing-cpu-usage[Optimizing CPU usage with mount namespace encapsulation].
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 3]
```diff
@@ -884,6 +884,55 @@ In {product-title} {product-version}, you can use the {factory-prestaging-tool}
...
+You can use the `ClusterLabelSelector` field for greater flexibility in selecting clusters for update.
+You can use timeout settings to determine what happens if an update fails for a cluster, for example, skipping the failing cluster and continuing to upgrade other clusters, or stopping policy remediation for all clusters. 
+For more information see xref:../scalability_and_performance/cnf-talm-for-cluster-upgrades.adoc[Topology Aware Lifecycle Manager for cluster updates].
+
+[id="scalability-and-performance-mount-namespace"]
+==== Mount namespace encapsulation (Technology Preview)
+
+Encapsulation is the process of moving all Kubernetes-specific mount points to an alternative namespace to reduce the visibility and performance impact of a large number of mount points in the default namespace. Previously, mount namespace encapsulation has been deployed transparently in OpenShift Container Platform specifically for Distributed Units (DUs) installed using the ZTP process. In OpenShift Container Platform v4.12, this proven functionality is being rolled out to all customers as a configurable option.
```
Comment (aireilly) [inline release_notes/ocp-4-12-release-notes.adoc:898]:
```suggestion
Encapsulation is the process of moving all Kubernetes-specific mount points to an alternative namespace to reduce the visibility and performance impact of a large number of mount points in the default namespace. Previously, mount namespace encapsulation has been deployed transparently in {product-title} specifically for Distributed Units (DUs) installed using GitOps ZTP. In {product-title} v4.12, this proven functionality is being rolled out to all customers as a configurable option.
```

**Comment by slovern** [comment]
@josclark42 can you rpovide QA ack for sectron on TALM -  [](https://issues.redhat.com/browse/TELCODOCS-477)
- from line 890

**Comment by slovern** [comment]
https://issues.redhat.com/browse/TELCODOCS-961 
@mcornea can you provide QA ack for user override of policy intervals (starts at [id="ocp-4-12-ztp-config-policy-eval-intervals"] )?

**Comment by slovern** [comment]
TELCODOCS-468
@yliu127 can you provide QA ack for "Extedn ZTP functionality" (starts at [id="ocp-4-12-ztp-sno-workload-partitioning"] )

**Comment by mcornea** [comment]
> https://issues.redhat.com/browse/TELCODOCS-961 @mcornea can you provide QA ack for user override of policy intervals (starts at [id="ocp-4-12-ztp-config-policy-eval-intervals"] )?

lgtm

**Comment by slovern** [comment]
TELCODOCS-469
@yliu127 can you provide QA ack for "GitOps ZTP " 
Two entries:
- [id="ocp-4-12-ztp-hub-templates"]
- [id="ocp-4-12-ztp-argocd-managed-cluster-limits"]

**Comment by slovern** [comment]
TELCODOCS-640 
@sakhoury
@djuran
A QA approval is needed here. @tmulquee is out of office so I've picked your names form his original PR.
If one of you is a QA can you take a look at the section on Mount namespace encapsulation?
relevant section is [id="scalability-and-performance-mount-namespace"]

**Comment by Bonnie-Block** [comment]
> [TELCODOCS-640](https://issues.redhat.com//browse/TELCODOCS-640) @sakhoury @djuran A QA approval is needed here. @tmulquee is out of office so I've picked your names form his original PR. If one of you is a QA can you take a look at the section on Mount namespace encapsulation? relevant section is [id="scalability-and-performance-mount-namespace"]

/lgtm

File: release_notes/ocp-4-12-release-notes.adoc (hunk 1 above)
Comment (slovern) [reply release_notes/ocp-4-12-release-notes.adoc]: done

File: release_notes/ocp-4-12-release-notes.adoc (hunk 2 above)
Comment (slovern) [reply release_notes/ocp-4-12-release-notes.adoc:902]: Thanks - done

File: release_notes/ocp-4-12-release-notes.adoc (hunk 3 above)
Comment (slovern) [reply release_notes/ocp-4-12-release-notes.adoc:898]: thanks. Done

**Comment by slovern** [comment]
@Bonnie-Block 
would you be able to look at two additional changes


    [id="ocp-4-12-ztp-hub-templates"]
    [id="ocp-4-12-ztp-argocd-managed-cluster-limits"]

**Comment by Bonnie-Block** [comment]
> 

/lgtm

File: release_notes/ocp-4-12-release-notes.adoc [hunk 4]
```diff
@@ -1010,6 +1073,15 @@ If you have Operator projects that were previously created or maintained with Op
...
 
+[discrete]
+[id="ocp-4-12-logical-volume-manager-storage"]
+=== LVM Operator is now called Logical Volume Manager Storage
+
+The LVM Operator that was previously delivered with {rh-storage-first} requires installation through the {rh-storage}.
+In {product-title} v4.12, the LVM Operator has been renamed _Logical Volume Manager Storage_.
+Now, you install it as a standalone Operator from the OpenShift Operator catalog.
+Logical Volume Manager Storage provides dynamic provisioning of block storage on a single, limited resources {sno} cluster.
+
```
Comment (shyRozen) [inline release_notes/ocp-4-12-release-notes.adoc:1084]: LGTM

File: release_notes/ocp-4-12-release-notes.adoc [hunk 5]
```diff
@@ -170,6 +170,19 @@ If a third-party CSI driver is present in the cluster, {product-title} does not
 Components with versions earlier than those above are still supported, but are deprecated. These versions are still fully supported, but version 4.12 of {product-title} requires vSphere virtual hardware version 15 or later. For more information, see xref:../release_notes/ocp-4-12-release-notes#ocp-4-12-deprecated-removed-features[Deprecated and removed features].
 
 Failing to meet the above requirements prevents {product-title} from upgrading to {product-title} 4.13 or later.
+[id="ocp-4-12-cluster-capabilities"]
```
Comment (slovern) [inline release_notes/ocp-4-12-release-notes.adoc:173]: Approvalss for this change are in PR https://github.com/openshift/openshift-docs/pull/54585/

**Comment by slovern** [comment]
/ label telco

File: release_notes/ocp-4-12-release-notes.adoc [hunk 6]
```diff
@@ -1011,6 +1024,54 @@ KMM includes the following features for connected environments only:
...
+
+For more information, see xref:../scalability_and_performance/optimizing-cpu-usage.adoc#optimizing-cpu-usage[Optimizing CPU usage with mount namespace encapsulation].
+
+[id="ocp-4-12-ztp-sno-workload-partitioning"]
+==== Changing the workload partitioning CPU set in {sno} clusters that are deployed with GitOps ZTP
+
+You can configure the workload partitioning CPU set in {sno} clusters that you deploy with GitOps ZTP.
+To do this, you specify cluster management CPU resources with the `cpuset` field of the `SiteConfig` custom resource (CR) and the `reserved` field of the group `PolicyGenTemplate` CR.
+The value that you set for `cpuset` should match the value set in the cluster `PerformanceProfile` CR `.spec.cpu.reserved` field for workload partitioning.
```
Comment (yliu127) [inline release_notes/ocp-4-12-release-notes.adoc:1049]: /lgtm

File: release_notes/ocp-4-12-release-notes.adoc [hunk 7]
```diff
@@ -1011,6 +1024,54 @@ KMM includes the following features for connected environments only:
 ==== Hub and spoke cluster support (Technology Preview)
 
 For hub and spoke deployments in an environment that can access the internet, you can use the kernel module management (KMM) Operator deployed in the hub cluster to manage the deployment of the required kernel modules to one or more managed clusters. 
+[id="scalability-and-performance-talm-updates"]
+==== {cgu-operator-first}
+
+{cgu-operator-first} now provides more detailed status information and messages, and redesigned conditions. 
+You can use the `ClusterLabelSelector` field for greater flexibility in selecting clusters for update.
```
Comment (mburke5678) [inline release_notes/ocp-4-12-release-notes.adoc:1031]: Maybe put _for example_ at the beginning for clarity.

File: release_notes/ocp-4-12-release-notes.adoc [hunk 8]
```diff
@@ -1011,6 +1024,54 @@ KMM includes the following features for connected environments only:
...
+You can use timeout settings to determine what happens if an update fails for a cluster, for example, skipping the failing cluster and continuing to upgrade other clusters, or stopping policy remediation for all clusters. 
+For more information see xref:../scalability_and_performance/cnf-talm-for-cluster-upgrades.adoc[Topology Aware Lifecycle Manager for cluster updates].
+
+[id="scalability-and-performance-mount-namespace"]
+==== Mount namespace encapsulation (Technology Preview)
+
+Encapsulation is the process of moving all Kubernetes-specific mount points to an alternative namespace to reduce the visibility and performance impact of a large number of mount points in the default namespace. Previously, mount namespace encapsulation has been deployed transparently in {product-title} specifically for Distributed Units (DUs) installed using GitOps ZTP. In {product-title} v4.12, this proven functionality is now available as a configurable option.
```
Comment (mburke5678) [inline release_notes/ocp-4-12-release-notes.adoc:1038]: s/{product-title}/{product-title},

I might remove _proven_. It sounds to me like marketing-speak.

File: release_notes/ocp-4-12-release-notes.adoc [hunk 9]
```diff
@@ -1011,6 +1024,54 @@ KMM includes the following features for connected environments only:
...
+The value that you set for `cpuset` should match the value set in the cluster `PerformanceProfile` CR `.spec.cpu.reserved` field for workload partitioning.
+
+For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.adoc#ztp-using-hub-cluster-templates_ztp-advanced-policy-config[Using hub templates in PolicyGenTemplate CRs].
+
+[id="ocp-4-12-ztp-hub-templates"]
+==== RHACM hub template functions now available for use with GitOps ZTP
+
+Hub template functions are now available for use with GitOps ZTP using {rh-rhacm-first} and {cgu-operator-first}.
+Hub-side cluster templates reduce the need to create separate policies for many clusters with similiar configurations but with different values.
+For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.adoc#ztp-using-hub-cluster-templates-with-talm_ztp-advanced-policy-config[Using hub templates in PolicyGenTemplate CRs].
```
Comment (mburke5678) [inline release_notes/ocp-4-12-release-notes.adoc:1058]: I think this link is wrong. It is going to the top of the assembly, not the module. 
Also, should this link to the same module as the previous link?

File: release_notes/ocp-4-12-release-notes.adoc [hunk 10]
```diff
@@ -2318,6 +2388,16 @@ In the following tables, features are marked with the following statuses:
 |Not Available
 |Technology Preview
 
+|{cgu-operator}
```
Comment (mburke5678) [inline release_notes/ocp-4-12-release-notes.adoc:2391]: Should this be `cgu-operator-first`, where it has been so long since the full name was mentioned?

**Comment by mburke5678** [comment]
@slovern A few nits. One bad link. Otherwise LGTM

File: release_notes/ocp-4-12-release-notes.adoc (hunk 8 above)
Comment (slovern) [reply release_notes/ocp-4-12-release-notes.adoc:1038]: True. I've removed it.

File: release_notes/ocp-4-12-release-notes.adoc (hunk 10 above)
Comment (slovern) [reply release_notes/ocp-4-12-release-notes.adoc:2391]: yes, I agree.

File: release_notes/ocp-4-12-release-notes.adoc (hunk 9 above)
Comment (aireilly) [reply release_notes/ocp-4-12-release-notes.adoc:1058]: Yes - use `../scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.adoc#ztp-using-hub-cluster-templates_ztp-advanced-policy-config`

File: release_notes/ocp-4-12-release-notes.adoc (hunk 10 above)
Comment (slovern) [reply release_notes/ocp-4-12-release-notes.adoc:2391]: makes sense, yes.

File: release_notes/ocp-4-12-release-notes.adoc [hunk 11]
```diff
@@ -1011,6 +1024,54 @@ KMM includes the following features for connected environments only:
...
+For more information, see xref:../scalability_and_performance/optimizing-cpu-usage.adoc#optimizing-cpu-usage[Optimizing CPU usage with mount namespace encapsulation].
+
+[id="ocp-4-12-ztp-sno-workload-partitioning"]
+==== Changing the workload partitioning CPU set in {sno} clusters that are deployed with GitOps ZTP
+
+You can configure the workload partitioning CPU set in {sno} clusters that you deploy with GitOps ZTP.
+To do this, you specify cluster management CPU resources with the `cpuset` field of the `SiteConfig` custom resource (CR) and the `reserved` field of the group `PolicyGenTemplate` CR.
+The value that you set for `cpuset` should match the value set in the cluster `PerformanceProfile` CR `.spec.cpu.reserved` field for workload partitioning.
+
+For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.adoc#ztp-using-hub-cluster-templates_ztp-advanced-policy-config[Using hub templates in PolicyGenTemplate CRs].
```
Comment (aireilly) [inline release_notes/ocp-4-12-release-notes.adoc:1051]: For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-reference-cluster-configuration-for-vdu.adoc#ztp-sno-du-enabling-workload-partitioning_sno-configure-for-vdu[Workload partitioning].

File: release_notes/ocp-4-12-release-notes.adoc (hunk 7 above)
Comment (mburke5678) [reply release_notes/ocp-4-12-release-notes.adoc:1130]: Sorry! I read _for update_ as _for example_.

**Pull request merged**